packages/discovery-react-components/src/components/TooltipHighlight/TooltipHighlight.tsx

```diff
@@ -0,0 +1,164 @@
+      }}
+    >
+      <Tooltip
+        autoOrientation={true}
+        tabIndex={0}
+        showIcon={false}
+        open={tooltipInfo.isOpen}
+        triggerText={
+          <div
+            style={{
+              // Provide size so that tooltip knows boundry where to draw the tooltip
+              width: tooltipInfo.rectTooltipArea.width,
+              height: tooltipInfo.rectTooltipArea.height,
+              pointerEvents: 'none'
+            }}
+          />
+        }
+        children={<div>{tooltipInfo.tooltipContent}</div>}
+      />
+    </div>
+  );
+};
+
+export function calcToolTipContent(
+  facetInfoMap: FacetInfoMap,
+  facetId: string,
+  enrichValue: string
+) {
+  const tableContent = [];
+  let enrichColor = '';
+  let enrichFacetDisplayname = '';
+  if (facetInfoMap[facetId]) {
+    enrichColor = facetInfoMap[facetId].color;
+    enrichFacetDisplayname = facetInfoMap[facetId].displayName;
+    // Will have multiple entries after overlapping is implemented
+    tableContent.push({
+      enrichColor: enrichColor,
+      enrichFacetDisplayname: ellipsisMiddle(enrichFacetDisplayname),
+      enrichValue: ellipsisMiddle(enrichValue)
+    });
+  }
+
+  let tooltipContent = undefined;
+
+  if (enrichFacetDisplayname || enrichValue) {
+    tooltipContent = (
+      <div className={cx(baseTooltipCustomContent)} data-testid="tooltip_highlight_content">
+        <div className={cx(baseTooltipContentHeader)}>
+          {defaultMessages.enrichmentsHeaderLabel} ({tableContent.length})
+        </div>
+        <table>
+          {tableContent.map((oneRow, index) => {
+            let rowBorderClass = {};
+            if (index < tableContent.length - 1) {
+              rowBorderClass = {
+                borderBottom: `1px solid #7A7979`
+              };
+            }
+            return (
+              <tr>
+                <td className={cx(baseTooltipContentCell)} style={rowBorderClass}>
+                  <div
+                    className={cx(baseTooltipBoxColor)}
+                    style={{
+                      backgroundColor: oneRow.enrichColor
+                    }}
+                  />
+                </td>
+                <td className={cx(baseTooltipContentCell)} style={rowBorderClass}>
+                  <span className={cx(baseTooltipContentCellBuffer)}>
+                    {oneRow.enrichFacetDisplayname}
+                  </span>
+                </td>
+                <td className={cx(baseTooltipContentCell)} style={rowBorderClass}>
+                  {oneRow.enrichValue &&
+                    oneRow.enrichValue.localeCompare(oneRow.enrichFacetDisplayname) !== 0 &&
+                    `${oneRow.enrichValue}`}
+                </td>
+              </tr>
+            );
+          })}
+        </table>
+      </div>
+    );
+  }
+  return tooltipContent;
+}
+
+function ellipsisMiddle(text: string) {
+  const MAX_CONTENT_LENGTH = 30; // even number
+  const ELLIPSIS = '...';
```

Comment:
Ellipsis is a unicode character: https://www.compart.com/en/unicode/U+2026. Use that instead of 3 periods.